Add annotation on hpa to disable ownership validation

junekhan opened the pull request:
<!-- Thank you for contributing!

     Read more about how you can contribute in our contribution guide:
     https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md
-->

Add annotation `validations.keda.sh/shared-ownership` on hpa to disable ownership validation

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

<!--
  Make sure to link the related issue for this change
  If it requires multiple PRs and/or a PR on another repo as well, please use "Relates to" instead.
-->
Fixes #5516 

<!--
  Make sure to link the related PRs for changes such as documentation & Helm charts
-->
Relates to #

zroubalik reviewed (commented):
I would probably name this annotation `validations.keda.sh/hpa-ownership` or something around that, to make it more clear what is it about.

WDYT @kedacore/keda-core-contributors

tomkerkhove commented:
Agreed!

zroubalik commented:
/run-e2e internal

zroubalik reviewed (commented):
@junekhan could you please fix the conflict in the changelog and also document this new annotation? https://keda.sh/docs/2.13/concepts/scaling-deployments/

junekhan commented:
> @junekhan could you please fix the conflict in the changelog and also document this new annotation? https://keda.sh/docs/2.13/concepts/scaling-deployments/

I've fixed the conflict and updated the document for v2.13.
